### PR TITLE
Fix Concurrent Snapshot Repository Remove Issues

### LINKED ISSUE #74858: Concurrent Snapshot Delete and Snapshot Repository Removal can Lead to Master Nodes Looping
When executing a snapshot delete we do not check if the repository we are deleting from actually still exists when attempting to put the delete in the cluster state. 
In older versions pre-7.8 where we found this, this would lead to a snapshot delete operation that would apply to a repository that is gone from the cluster state. In versions newer than 7.8 that make use of https://github.com/elastic/elasticsearch/pull/55773 this may lead to an infinite loop on the master because the consistent state update will never find the metadata expected by the repository instance in the cluster state.

Working on a fix.

### PR DESCRIPTION
**Pull request by original-brownbear**
We need to fail repository operations as soon as a repository has been shut down
to avoid getting into broken cluster state situations where operations are created
in the cluster state but the repository they target was removed concurrently.

closes #74858

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java [hunk 1]
```diff
@@ -1221,6 +1223,45 @@ public void testGetReposWithWildcard() {
...
+            deleteFutures.add(clusterAdmin().prepareDeleteSnapshot(repoName, snapshotName).execute());
+        }
+
+        try {
+            assertAcked(deleteRepoFuture.actionGet());
+        } catch (Exception e) {
+            assertThat(
+                e.getMessage(),
+                containsString(
+                    "trying to modify or unregister repository [test-repo] that is currently " + "used (snapshot deletion is in progress)"
```
Comment (DaveCTurner) [inline server/src/internalClusterTest/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java:1249]: nit: evaluate this string constant?

**Review by DaveCTurner — COMMENTED** [review]
Certainly seems to fix it, but the description of https://github.com/elastic/elasticsearch/issues/74858 indicates that the problem is that we allow an orphaned delete to be added to the cluster state so I'm wondering why we aren't addressing this in the cluster state updates instead (perhaps as well).

Put differently, it's unclear to me that checking the repo lifecycle in all these places is a sufficient fix. The repo is stopped by a cluster state update, whereas these checks are on snapshot threads, so in principle the lifecycle can change independently of these checks.

**Review by DaveCTurner — APPROVED** [review]
LGTM :)

**Review by tlrx — APPROVED** [review]
LGTM bis

**Comment by original-brownbear** [comment]
Thanks David and Tanguy!

**Pull request merged**